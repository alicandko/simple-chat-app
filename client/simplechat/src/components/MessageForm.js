import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './MessageForm.scss';
export class MessageForm extends Component {
  static props = {
    username: PropTypes.string.isRequired,
    messageSend: PropTypes.func.isRequired,
  };

  state = {
    messageText: null,
  };

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.messageSend({
      username: this.props.username,
      text: this.state.messageText,
    });
  };

  handleChange = (event) => {
    this.setState({
      messageText: event.target.value,
    });
  };

  render() {
    const messageText = this.props.messageText;
    return (
      <form className="message-form form" onSubmit={this.handleSubmit}>
        <input
          type="text"
          className="message-form__message-input"
          onChange={this.handleChange}
          value={messageText}
        />
        <input
          type="submit"
          className="message-form__button"
          disabled={!this.state.messageText}
          value="Send"
        />
      </form>
    );
  }
}
