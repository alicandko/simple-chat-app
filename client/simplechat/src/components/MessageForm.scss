.message-form {
  margin: 10px;
  display: flex;
  flex-direction: row;

  &__button {
    margin-left: 10px;
    flex: 0 1;
  }
}
