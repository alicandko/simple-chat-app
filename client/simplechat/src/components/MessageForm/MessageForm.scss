.message-form {
  margin: 10px;
  display: flex;
  flex-direction: row;

  &.form {
    border-top: none;
  }

  &__button {
    margin-left: 10px;
    flex: 0 1;
  }
}
