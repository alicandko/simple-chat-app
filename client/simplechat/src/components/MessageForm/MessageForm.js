import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './MessageForm.scss';

export class MessageForm extends Component {
  static props = {
    messageSend: PropTypes.func.isRequired,
    username: PropTypes.string.isRequired,
  };

  state = {
    messageText: '',
  };

  handleSubmit = (event) => {
    event.preventDefault();

    this.props.messageSend({
      username: this.props.username,
      text: this.state.messageText,
    });

    this.setState({
      messageText: '',
    });
  };

  handleChange = (event) => {
    this.setState({
      messageText: event.target.value,
    });
  };

  render() {
    const messageText = this.state.messageText;
    return (
      <form className="message-form form" onSubmit={this.handleSubmit}>
        <input
          type="text"
          className="message-form__message-input text-input"
          onChange={this.handleChange}
          value={messageText}
        />
        <input
          type="submit"
          className="message-form__button button"
          disabled={!this.state.messageText}
          value="Send"
        />
      </form>
    );
  }
}
