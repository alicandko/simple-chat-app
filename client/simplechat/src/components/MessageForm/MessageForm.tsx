import React, { Component } from 'react';
import { OutgoingMessage } from '../../redux/types';

import './MessageForm.scss';

interface Props {
  messageSend: (message: OutgoingMessage) => void;
  username: string;
}

interface State {
  messageText: string;
}

export class MessageForm extends Component<Props, State> {
  public state = {
    messageText: '',
  };

  public handleSubmit = (event) => {
    event.preventDefault();

    this.props.messageSend({
      text: this.state.messageText,
      username: this.props.username,
    });

    this.setState({
      messageText: '',
    });
  };

  public handleChange = (event) => {
    this.setState({
      messageText: event.target.value,
    });
  };

  public render() {
    const messageText = this.state.messageText;
    return (
      <form className="message-form form" onSubmit={this.handleSubmit}>
        <input
          type="text"
          className="message-form__message-input text-input"
          onChange={this.handleChange}
          value={messageText}
        />
        <input
          type="submit"
          className="message-form__button button"
          disabled={!this.state.messageText}
          value="Send"
        />
      </form>
    );
  }
}
