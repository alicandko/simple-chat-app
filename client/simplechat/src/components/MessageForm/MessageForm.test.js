import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MessageForm } from './MessageForm';

const mockFunction = jest.fn();

describe('MessageForm', () => {
  it('should render correctly', () => {
    const component = renderer
      .create(<MessageForm username="" messageSend={mockFunction} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should have the correct passed props', () => {
    const wrapper = mount(<MessageForm username="mockUsername" messageSend={mockFunction} />);

    expect(wrapper.prop('username')).toEqual('mockUsername');
    expect(wrapper.prop('messageSend')).toEqual(mockFunction);
  });

  describe('when button is clicked', () => {
    it('should execute the messageSend function', () => {
      const wrapper = mount(
        <MessageForm username="mockUsername" connectRequested={mockFunction} />,
      );

      wrapper.find('.message-form__button').simulate('click');

      expect(mockFunction.mock.calls.length).toEqual(0);
    });
  });

  describe('when enter is pressed', () => {
    it('should execute the messageSend function', () => {
      const wrapper = mount(
        <MessageForm username="mockUsername" connectRequested={mockFunction} />,
      );

      wrapper.find('.message-form__button').simulate('keypress', { key: 'Enter' });

      expect(mockFunction.mock.calls.length).toEqual(0);
    });
  });
});
