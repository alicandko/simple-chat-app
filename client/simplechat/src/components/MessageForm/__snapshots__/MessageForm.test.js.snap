// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm should render correctly 1`] = `
<form
  className="message-form form"
  onSubmit={[Function]}
>
  <input
    className="message-form__message-input"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    className="message-form__button"
    disabled={true}
    type="submit"
    value="Send"
  />
</form>
`;
